### PR TITLE
Alerts for WAL metrics

### PR DESCRIPTION
**Pull request by codesome**

### CODE REVIEW

**Comment by codesome** [comment]
I tried to take care of it with the range vector (but things like `CortexWALCorruption` should be immediately alerted). I was having a hard time figuring out a combination of functions and `for` period. 

I will take a dig again to use the `for` period.

File: cortex-mixin/alerts.libsonnet [hunk 1]
```diff
@@ -226,6 +226,81 @@ local windows = [
...
+              message: |||
+                {{ $labels.namespace }}/{{ $labels.instance }} failed to delete checkpoint.
+              |||,
+            },
+          },
+          {
+            // 2 or more failed checkpoint deletion in 1h means something is wrong.
+            alert: 'CortexCheckpointDeletionFailed',
+            expr: |||
+              increase(cortex_ingester_checkpoint_deletions_failed_total[1h]) > 1
```
Comment (gouthamve) [inline cortex-mixin/alerts.libsonnet:293]: Can we make this `2h`?

File: cortex-mixin/alerts.libsonnet [hunk 2]
```diff
@@ -226,6 +226,81 @@ local windows = [
               |||,
             },
           },
+          {
+            // Alert immediately if WAL is corrupt.
+            alert: 'CortexWALCorruption',
+            expr: |||
+              rate(cortex_ingester_wal_corruptions_total[5m]) > 0
```
Comment (gouthamve) [inline cortex-mixin/alerts.libsonnet:233]: Why are we using rate here and increase everywhere else?

Comment (codesome) [reply cortex-mixin/alerts.libsonnet:233]: Does not matter actually, I will make it increase to be consistent

**Pull request merged**